### PR TITLE
Test results filters on role, date range

### PR DESCRIPTION
**Pull request by jeremyzitomer-usds**
## Related Issue or Background Info

- Date range filter: https://app.zenhub.com/workspaces/simplereport-better-data-team-605b43d51ea9f700119a48ee/issues/cdcgov/prime-simplereport/1029
- Role filter: https://app.zenhub.com/workspaces/simplereport-better-data-team-605b43d51ea9f700119a48ee/issues/cdcgov/prime-simplereport/1113

## Changes Proposed

- Add front-end components to filter test result search on date range and person role
- Updated `testResults` and `testResultsCount` endpoints to take optional filter parameters for date range and person role

## Screenshots / Demos
https://user-images.githubusercontent.com/76444566/122777565-272c7480-d27a-11eb-9ae4-d60bff9528be.mov
https://user-images.githubusercontent.com/76444566/122777577-2a276500-d27a-11eb-826a-3b31f12b4904.mov

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java [hunk 1]
```diff
@@ -126,24 +135,112 @@ public TestOrderService(
...
+                ? "false"
+                : symptomatic == OptionalBoolean.NO
+                    ? "true"
+                    : symptomatic == OptionalBoolean.WOULD_NOT_SPECIFY ? "false" : "false";
+        p =
+            cb.and(
+                p,
+                cb.equal(
+                    cb.function(
+                        "jsonb_extract_path_text",
```
Comment (timothybest-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java:153]: I believe filtering on a `jsonb` field will cause performance issue in production

Comment (timothybest-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java:153]: I would be interested in seeing what the plan is for the query that is generated. You should be able to grab the query from the console logs and then run it manually against your local DB with `explain`

**Review by hmyers-cms — APPROVED** [review]
LGTM

**Review by emmastephenson — APPROVED** [review]

**Comment by jeremyzitomer-usds** [comment]
@emmastephenson has approved BE, @hmyers-cms has approved design, now just need a FE approval!

**Review by nickclyde — APPROVED** [review]
This all looks good to me, great work Jeremy!

**Pull request merged**